src/ucp/core/ucp_worker.c

```diff
@@ -451,8 +451,23 @@ ucs_status_t ucp_worker_create(ucp_context_h context, ucs_thread_mode_t thread_m
                            sizeof(*worker->ep_config) * config_count,
                         "ucp worker");
     if (worker == NULL) {
-        status = UCS_ERR_NO_MEMORY;
-        goto err;
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    if (thread_mode != UCS_THREAD_MODE_MULTI) {
+        worker->mt_lock.mt_type = UCP_MT_TYPE_NONE;
+    } else if (context->config.ext.use_mt_mutex) {
+        worker->mt_lock.mt_type = UCP_MT_TYPE_MUTEX;
+    } else {
+        worker->mt_lock.mt_type = UCP_MT_TYPE_SPINLOCK;
+    }
+
+    UCP_THREAD_LOCK_INIT_CONDITIONAL(&worker->mt_lock);
+
+    if (UCP_THREAD_IS_REQUIRED(&worker->mt_lock)) {
+        worker->thread_mode = thread_mode;
```

Comment:
can we check UCP_THREAD_IS_REQUIRED(&worker->mt_lock) inside ucp_worker_query, instead of keeping the field worker->thread_mode